### PR TITLE
feat: bump SDK to 0.3.0

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

* Bumps SDK to `0.3.0`
* Adds error codes to Connector function
* Adds error handling to element template
* Adjusts validation annotations from `jakarta.validation` to `javax.validation`

## Related issues

related to https://github.com/camunda/team-connectors/issues/203

### CODE REVIEW

**Comment by tmetzke** [comment]
I will add this to the element template as well

**Review by igpetrov — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
     <!-- connector SDK version -->
-    <version.connector-core>0.2.2</version.connector-core>
+    <version.connector-core>0.3.0</version.connector-core>
```
Comment (tmetzke) [inline pom.xml:17]:
```suggestion
    <version.connector-core>0.3.0.alpha2</version.connector-core>
```

File: pom.xml [hunk 2]
```diff
@@ -14,7 +14,7 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
     <!-- connector SDK version -->
-    <version.connector-core>0.2.2</version.connector-core>
+    <version.connector-core>0.3.0.alpha2</version.connector-core>
```
Comment (tmetzke) [inline pom.xml:17]:
```suggestion
    <version.connector-core>0.3.0-alpha2</version.connector-core>
```

File: pom.xml [hunk 3]
```diff
@@ -14,7 +14,7 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
     <!-- connector SDK version -->
-    <version.connector-core>0.2.2</version.connector-core>
+    <version.connector-core>0.3.0-alpha2</version.connector-core>
```
Comment (tmetzke) [inline pom.xml:17]:
```suggestion
    <version.connector-core>0.3.0</version.connector-core>
```

**Comment by tmetzke** [comment]
Runs successfully with `0.3.0-alpha2`: https://github.com/camunda/connector-template/actions/runs/3369381402

🚨 This will be merged when SDK v`0.3.0` is available.

**Review by igpetrov — APPROVED** [review]

**Pull request merged**